flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/BigQuerySource.java

```diff
@@ -351,14 +352,42 @@ public static BigQuerySource<GenericRecord> streamAvros(BigQueryReadOptions read
                                                         "%s.%s.%s",
                                                         connectOptions.getProjectId(),
                                                         connectOptions.getDataset(),
-                                                        connectOptions.getTable()),
+                                                        sanitizeAvroSchemaName(
+                                                                connectOptions.getTable())),
                                                 tableSchema.getFields())
                                         .toString()))
                 .setReadOptions(readOptions)
                 .setSourceBoundedness(Boundedness.CONTINUOUS_UNBOUNDED)
                 .build();
     }
 
+    /** Replaces invalid characters with underscore. */
+    private static String sanitizeAvroSchemaName(String name) {
```

Comment:
Is there any specific case that fails or causes an issue?

Reply:
If source table name has hyphen, then avro schema name validation will fail. It only accepts alphabets, digits and underscore. Found this while fixing the BQ table names in our e2e tests.

Reply:
Since the avro schema name is completely internal and hidden from the user, it doesnt matter how we sanitize. We just need to make sure that a valid BQ table name is not rejected by the connector because of some internal avro standard.

Reply:
Note that this is on the source side only.